lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -1305,7 +1334,10 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
 
   if (EC2 != EC2End) {
     if (Context.Complain) {
-      Context.Diag2(D2->getLocation(), diag::warn_odr_tag_type_inconsistent)
+      Context.Diag2(D2->getLocation(),
+                    Context.ErrorOnTagTypeMismatch
+                        ? diag::err_odr_tag_type_inconsistent
+                        : diag::warn_odr_tag_type_inconsistent)
```

Comment:
Maybe use a small function or lambda for these repeated statements (get constant for error or warning). (But a separate is needed for every kind of error-warning.)

Reply:
Implemented a quick lookup method.